### PR TITLE
Configure permissionless builder in native and docker demo

### PR DESCRIPTION
**Pull request by sveitser**
A bunch of fixes to make the builder work.

### CODE REVIEW

File: builder/src/bin/permissionless-builder.rs [hunk 1]
```diff
@@ -25,12 +24,11 @@ pub struct NonPermissionedBuilderOptions {
     /// URL of the HotShot DA web server.
```
Comment (nyospe) [inline builder/src/bin/permissionless-builder.rs:24]: This comment should also be updated. @move47

**Review by nyospe — COMMENTED** [review]
No programatic issues. A naming quibble and a stale comment.

File: builder/src/bin/permissionless-builder.rs [hunk 2]
```diff
@@ -25,12 +24,11 @@ pub struct NonPermissionedBuilderOptions {
...
     /// The builder will subscribe to this server to receive hotshot events
     #[clap(
-        short,
         long,
-        env = "ESPRESSO_SEQUENCER_DA_SERVER_URL",
+        env = "ESPRESSO_BUILDER_SEQUENCER_URL",
```
[the remark this answers is not in the document]
Comment (sveitser) [reply builder/src/bin/permissionless-builder.rs:28]: >  Is this the URL of an espresso sequencer node that is a DA committee member and provides the hotshot_event_service?

Yes, think so.

[the remark this answers is not in the document]
Comment (sveitser) [reply builder/src/bin/permissionless-builder.rs:28]: The explicit `ESPRESSO_SEQUENCER_HOTSHOT_EVENT_STREAMING_API_URL` is probably best.

File: builder/src/bin/permissionless-builder.rs (hunk 1 above)
Comment (sveitser) [reply builder/src/bin/permissionless-builder.rs]: https://github.com/EspressoSystems/espresso-sequencer/pull/1324/commits/8ed6b1806a1cccc8981efce30fbb888e3bc14882

File: builder/src/bin/permissionless-builder.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (sveitser) [reply builder/src/bin/permissionless-builder.rs]: https://github.com/EspressoSystems/espresso-sequencer/pull/1324/commits/8ed6b1806a1cccc8981efce30fbb888e3bc14882

**Review by tbro — APPROVED** [review]

**Pull request merged**